### PR TITLE
[FEATURE] Implementing the BlockedUsers Data Model

### PR DESCRIPTION
**Pull request by TheClashster**
Please prefix your pull request with one of the following: **[FEATURE]** **[FIX]** **[IMPROVEMENT]**.

**In raising this pull request, I confirm the following (please check boxes):**

- [x] I have read and understood the [contributors guide](https://github.com/CCExtractor/sample-platform/blob/master/.github/CONTRIBUTING.md).
- [x] I have checked that another pull request for this purpose does not exist.
- [x] I have considered, and confirmed that this submission will be valuable to others.
- [x] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [x] I give this submission freely, and claim no ownership to its content.

**My familiarity with the project is as follows (check one):**

- [ ] I have never used the project.
- [x] I have used the project briefly.
- [ ] I have used the project extensively, but have not contributed previously.
- [ ] I am an active contributor to the project.

---

Have made it pull request specific, by comparing the ID of the user with the one stored in the data model

### CODE REVIEW

File: mod_ci/controllers.py [hunk 1]
```diff
@@ -444,52 +445,39 @@ def kvm_processor(db, kvm_name, platform, repository, delay):
...
+        "%/CCExtractor/ccextractor.git")).first()
     if test_type == TestType.pull_request:
         branch = "pull_request"
-
-    linux_test = Test(TestPlatform.linux, test_type, fork.id, branch, commit, pr_nr)
-    db.add(linux_test)
-    windows_test = Test(TestPlatform.windows, test_type, fork.id, branch, commit, pr_nr)
-    db.add(windows_test)
+    # Create Linux test entry
+    linux = Test(TestPlatform.linux, test_type, fork.id, branch, commit,
```
Comment (satyammittal) [inline mod_ci/controllers.py]: You don't have to short each line. You can kept line length upto 90 characters. It will be easy to read.

Comment (TheClashster) [reply mod_ci/controllers.py]: Are you sure? The PEP8 guidelines say 79 characters.

Comment (satyammittal) [reply mod_ci/controllers.py]: Yes but on sample platform we have increased it to 90 characters, you can simply try pushing the changes upto 90 characters line.

Comment (satyammittal) [reply mod_ci/controllers.py]: We already simplified all lines which are exceeding 79 characters in precious commit:
https://github.com/CCExtractor/sample-platform/commit/8403bae4583da989a96633ef6c05844ae6e5ecb1

Comment (TheClashster) [reply mod_ci/controllers.py]: Will send another commit to that accord😃

**Comment by satyammittal** [comment]
Better be just undo all the changes of line length you have done in your commit. That is not useful.

File: mod_ci/controllers.py [hunk 2]
```diff
@@ -555,43 +555,48 @@ def start_ci():
...
-                        description="Tests canceled",
-                        context="CI - {name}".format(name=test.platform.value),
-                        target_url=url_for('test.by_id', test_id=test.id, _external=True)
-                    )
-            elif payload['action'] == 'reopened':
-                # Run tests again
-                queue_test(g.db, gh_commit, commit, TestType.pull_request, pr_nr=pr_nr)
+            # Check if user blacklisted
+            if BlockedUsers.query.filter(BlockedUsers.userID == payload[
+                   'pull_request']['user']['id']).first():
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: No trace besides a single log line, and user doesn't get notified? Please change.

Comment (TheClashster) [reply mod_ci/controllers.py]: Added return.

File: mod_ci/controllers.py [hunk 3]
```diff
@@ -555,43 +555,55 @@ def start_ci():
...
+                gh_commit.post(
+                    state=Status.ERROR,
+                    description="CI start aborted. You may be blocked from accesing this functionality",
+                    context="CI - {name}".format(name=platform_name),
+                    target_url=url_for('test.by_id', test_id=test_id, _external=True)
                     )
-            elif payload['action'] == 'reopened':
-                # Run tests again
-                queue_test(g.db, gh_commit, commit, TestType.pull_request, pr_nr=pr_nr)
+                return 'ERROR'
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: If you do a return here, you can drop the else and un-indent the entire other block. That'll also reduce the diff, and make it easier to maintain ;)

Comment (TheClashster) [reply mod_ci/controllers.py]: Will do 😃

File: mod_ci/controllers.py [hunk 4]
```diff
@@ -555,6 +555,17 @@ def start_ci():
 
         elif event == "pull_request":
             # If it's a PR, run the tests
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: Maybe move this comment to after your changes, that'll make more sense.

File: mod_ci/controllers.py [hunk 5]
```diff
@@ -555,6 +555,17 @@ def start_ci():
...
+            # Check if user blacklisted
+            if BlockedUsers.query.filter(BlockedUsers.userID == payload[
+                   'pull_request']['user']['id']).first():
+                log.critical("User Blacklisted")
+                gh_commit.post(
+                    state=Status.ERROR,
+                    description="CI start aborted. You may be blocked from accesing this functionality",
+                    context="CI - {name}".format(name=platform_name),
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: This name doesn't seem to be defined?

File: mod_ci/controllers.py (hunk 4 above)
Comment (TheClashster) [reply mod_ci/controllers.py]: One second.

File: mod_ci/controllers.py [hunk 6]
```diff
@@ -554,7 +554,17 @@ def start_ci():
...
         elif event == "pull_request":
-            # If it's a PR, run the tests
+            # Check if user blacklisted
+            if BlockedUsers.query.filter(BlockedUsers.userID == payload[
+                   'pull_request']['user']['id']).first():
+                log.critical("User Blacklisted")
+                gh_commit.post(
+                    state=Status.ERROR,
+                    description="CI start aborted. You may be blocked from accesing this functionality",
+                    target_url=url_for(@mod_ci.route('/'), _external=True)
```
Comment (canihavesomecoffee) [inline mod_ci/controllers.py]: No, that's not the correct way to link.

See https://github.com/CCExtractor/sample-platform/blob/8403bae4583da989a96633ef6c05844ae6e5ecb1/mod_auth/controllers.py#L53

**Comment by satyammittal** [comment]
@TheClashster Update your branch with latest changes

File: mod_ci/controllers.py [hunk 7]
```diff
@@ -564,6 +564,17 @@ def start_ci():
                 except KeyError:
                     g.log.critical("Didn't find a SHA value for a newly opened PR!")
                     g.log.debug(payload)
+                # Check if user blacklisted
+                if BlockedUsers.query.filter(BlockedUsers.userID == payload[
+                    'pull_request']['user']['id']).first() is not None:
+                    log.critical("User Blacklisted")
```
Comment (saurabhshri) [inline mod_ci/controllers.py]: I guess it'd be helpful to also log user ID here.

File: mod_ci/controllers.py [hunk 8]
```diff
@@ -564,6 +564,17 @@ def start_ci():
...
+                    gh_commit.post(
+                        state=Status.ERROR,
+                        description="CI start aborted. You may be blocked from accesing this functionality",
```
Comment (saurabhshri) [inline mod_ci/controllers.py]: Typo - "accessing".

Also maybe change it to something short like "Tests aborted - user is blacklisted.". What'd you think @canihavesomecoffee ?

Comment (TheClashster) [reply mod_ci/controllers.py]: This is the message posted to the user I think so this should do the job, will fix the typo. Thanks!

File: mod_ci/controllers.py (hunk 7 above)
Comment (canihavesomecoffee) [reply mod_ci/controllers.py]: Maybe, but as it won't occur a lot I don't see the need immediately.

Comment (canihavesomecoffee) [reply mod_ci/controllers.py]: And I'd rather not pollute the db with canceled tests of blocked users.

Comment (satyammittal) [reply mod_ci/controllers.py]: Github notification is enough, if someone ask, we could see that posting link.

**Pull request merged**